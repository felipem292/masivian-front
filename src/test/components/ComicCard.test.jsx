import React from 'react';
import { shallow } from 'enzyme';
const { ComicCard } = require("../../components/comicCard");
import '@testing-library/jest-dom';
import { useFetch } from '../../hooks/useFetch';
jest.mock('../../hooks/useFetch');

describe('unit test comic card', () => {
    
    test('should render correctly', () => {
        useFetch.mockReturnValue({ data: null, loading: true, error: null });
        const wrapper= shallow(<ComicCard/>);
        expect(wrapper).toMatchSnapshot();
    })

    test('should show card item', () => {
        const resp={
            img: "https://imgs.xkcd.com/comics/woodpecker.png", 
            title: "Woodpecker"
        };
        useFetch.mockReturnValue({ data: resp, loading: false, error: null });
        const wrapper= shallow(<ComicCard/>);
        const text= (wrapper.find('div').text());
        console.log(text);
        expect(wrapper).toMatchSnapshot();
        expect(text).not.toBe('loading....') ;
       
    })
    test('should show loading text', () => {
      
        useFetch.mockReturnValue({ data: null, loading: true, error: null });
        const wrapper= shallow(<ComicCard/>);
        const text= (wrapper.find('div').text());
        expect(text).toBe('loading....');       
       
    })
    
    
})
