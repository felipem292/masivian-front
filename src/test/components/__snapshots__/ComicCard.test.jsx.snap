// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit test comic card should render correctly 1`] = `
<Container
  fluid={false}
>
  <Row
    className="justify-content-md-center"
    noGutters={false}
  >
    <Col
      className="justify-content-md-center"
      lg={6}
      md={8}
      sm={12}
    >
      <div
        className="alert alert-info text-center"
      >
        loading....
      </div>
    </Col>
  </Row>
</Container>
`;

exports[`unit test comic card should show card item 1`] = `
<Container
  fluid={false}
>
  <Row
    className="justify-content-md-center"
    noGutters={false}
  >
    <Col
      className="justify-content-md-center"
      lg={6}
      md={8}
      sm={12}
    >
      <div
        className="card animate__animated animate__bounce justify-content-center"
      >
        <Cards
          img="https://imgs.xkcd.com/comics/woodpecker.png"
          title="Woodpecker"
        />
      </div>
    </Col>
  </Row>
</Container>
`;
